pjmedia/src/pjmedia/sdp.c

```diff
@@ -1165,22 +1167,30 @@ static void parse_media(pj_scanner *scanner, pjmedia_sdp_media *med,
     pj_scan_advance_n(scanner, 2, SKIP_WS);
 
     /* type */
-    pj_scan_get_until_ch(scanner, ' ', &med->desc.media);
-    pj_scan_get_char(scanner);
+    pj_scan_get(scanner, &cs_token, &med->desc.media);
+
+    if (pj_scan_get_char(scanner) != ' ') {
+        on_scanner_error(scanner);
+    }
 
     /* port */
     pj_scan_get(scanner, &cs_token, &str);
-    med->desc.port = (unsigned short)pj_strtoul(&str);
-    if (pj_scan_is_eof(scanner)) {
+    status = pj_strtoul3(&str, &num, 10);
+    if (status != PJ_SUCCESS || pj_scan_is_eof(scanner)) {
         on_scanner_error(scanner);
         return;
     }
+    med->desc.port = (unsigned short)num;
```

Comment:
do we need to guard against overflow?
also for the port number below.

Reply:
IMHO ideally yes, but not sure if it should be done in that parsing function as the SDP RFC does not seem to specify the limit. But our port data type is uint16.